pennylane/interfaces/tensorflow.py

```diff
@@ -184,3 +215,167 @@ def grad_fn(*dy, **tfkwargs):
+                vjps = _compute_vjp_new(dy, jacs, multi_measurements)
+
+            else:
+                # Need to compute the Jacobians on the backward pass (accumulation="backward")
+
+                if isinstance(gradient_fn, qml.gradients.gradient_transform):
+                    # Gradient function is a gradient transform.
+
+                    # Generate and execute the required gradient tapes
+                    if _n == max_diff or not context.executing_eagerly():
+
+                        with qml.tape.Unwrap(*tapes, params=params_unwrapped):
+                            vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+                                tapes,
+                                dy,
+                                gradient_fn,
+                                reduction=lambda vjps, x: vjps.extend(qml.math.unstack(x)),
+                                gradient_kwargs=gradient_kwargs,
+                            )
+
+                            vjps = processing_fn(execute_fn(vjp_tapes)[0])
+
+                    else:
+                        vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+                            tapes,
+                            dy,
+                            gradient_fn,
+                            reduction="extend",
+                            gradient_kwargs=gradient_kwargs,
+                        )
+
+                        # This is where the magic happens. Note that we call ``execute``.
+                        # This recursion, coupled with the fact that the gradient transforms
+                        # are differentiable, allows for arbitrary order differentiation.
+                        vjps = processing_fn(
+                            execute(
+                                vjp_tapes,
+                                device,
+                                execute_fn,
+                                gradient_fn,
+                                gradient_kwargs,
+                                _n=_n + 1,
+                                max_diff=max_diff,
+                            )
+                        )
+
+                else:
+                    # Gradient function is not a gradient transform
+                    # (e.g., it might be a device method).
+                    # Note that unlike the previous branch:
+                    #
+                    # - there is no recursion here
+                    # - gradient_fn is not differentiable
+                    #
+                    # so we cannot support higher-order derivatives.
+                    with qml.tape.Unwrap(*tapes, params=params_unwrapped):
+                        jac = gradient_fn(tapes, **gradient_kwargs)
+
+                    vjps = _compute_vjp_new(dy, jac, multi_measurements)
+
+            # filter out untrainable parameters if they happen to appear in the vjp
+            vjps = [vjp for vjp in vjps if 0 not in qml.math.shape(vjp)]

```

Comment:
I believe this is required when the gradient function is a device method and there are multiple tapes with different trainable parameters.

Reply:
Yes here the question is do we want to remove the internal stacking for backpropagation? Because you changed the tests to do the stacking on the user side. It sounds reasonable for me to remove it and have same behavior between backprop and gradient transform. But it should be done in another PR and also for autograd in my opinion. What do you think? @antalszava @eddddddy